klog.go

```diff
@@ -1647,3 +1647,22 @@ func KObjs(arg interface{}) []ObjectRef {
 	}
 	return objectRefs
 }
+
+// LogAs implements both the fmt.Stringer and logr.Marshal interface with some
+// custom functions. Plain text output then will include the result of the Text
+// function whereas structured output (such as JSON) will include the result of
+// the Object function.
+//
+// The same can be done with custom structs. The advantage of LogAs is that the
+// formatting code can be written inline in the same function that logs some
+// object.
+type LogAs struct {
```

Comment:
`LogAs` is a proposal. Alternative suggestions are welcome.

I find it sounds a bit too much like "kick ass"... :sweat_smile: 

Note that I chose to implement this as a struct. The advantage is that the reader of code which uses `LogAs` gets a hint what the functions do, at least when the struct gets initialized with named fields.

The disadvantage is that users might accidentally forget to initialize one of the functions. That compiles, but then will crash when used, at least as it stands now.

Reply:
What about just `As`?

Reply:
I like that proposal.

Reply:
I changed it.